### PR TITLE
gateway: open up CORS to everything (allow-origin *)

### PR DESCRIPTION
**Pull request by lgierth**
- [ ] needs testing on a gateway node.

### CODE REVIEW

**Comment by harlantwood** [comment]
Awesome!  So IPFS.io/IPFS/xxxyyyzzz will use this?

File: solarnet/roles/ipfs/templates/config.j2 [hunk 1]
```diff
@@ -51,30 +51,14 @@
...
       "Access-Control-Allow-Origin": [
-        "https://gateway.ipfs.io",
-        "https://ipfs.io",
-        "http://gateway.ipfs.io",
-        "http://ipfs.io",
-        "http://h.ipfs.io",
-        "http://localhost",
-        "http://localhost:8080",
-        "http://127.0.0.1",
-        "http://127.0.0.1:8080"
```
Comment (jbenet) [inline solarnet/roles/ipfs/templates/config.j2]: we should only change this for the `Gateway`, not the API -- actually, this should NOT have the public gateways, and only have the default parameters.

now it relies on one level of isolation: docker not rebinding the port. if it rebinds the port, then the api is usable outside.

Comment (lgierth) [inline solarnet/roles/ipfs/templates/config.j2]: sorry for the delay here --

> we should only change this for the Gateway, not the API

The API headers rules apply to `:8080/api`, which is a bit inconvenient here -- maybe we should just change that in go-ipfs?

> if port gets exposed to outer host -- docker firewall stopping?

No firewalling, I made docker and nginx expose :5001 and :8080 only to 127.0.0.1 and the cjdns VPN

Comment (jbenet) [inline solarnet/roles/ipfs/templates/config.j2]: Ooof this api/gateway duality in go-ipfs needs reworking.

**Comment by lgierth** [comment]
@harlantwood could you point me to something that should now work, i.e. with Allow-Origin: *

**Comment by harlantwood** [comment]
Hm, nothing I have up right now... But look forward to using this!

**Comment by harlantwood** [comment]
@lgierth looking good:

https://ipfs.io/ipfs/QmV8e64oRKabaPxgZBN8ojpQRovtbQ4E2LgUvGXhFkSqWf :--

**Comment by lgierth** [comment]
cool!

File: solarnet/roles/ipfs/templates/config.j2 (hunk 1 above)
Comment (lgierth) [inline solarnet/roles/ipfs/templates/config.j2]: Wild guess: ipfs/go-ipfs#2021 might fix this

**Pull request merged**